providers/io/socarxiv/normalizer.py

```diff
@@ -0,0 +1,112 @@
+from share.normalize import ctx
+from share.normalize.parsers import Parser
+from share.normalize import tools
+
+
+class Identifier(Parser):
+    url = ctx
+    base_url = tools.Static('https://osf.io/')
+
+
+class ThroughIdentifiers(Parser):
+    identifier = tools.Delegate(Identifier, ctx)
+
+
+class Person(Parser):
+    given_name = ctx.embeds.users.data.attributes.given_name
+    family_name = ctx.embeds.users.data.attributes.family_name
+    additional_name = ctx.embeds.users.data.attributes.middle_names
+    suffix = ctx.embeds.users.data.attributes.suffix
+    url = ctx.embeds.users.data.links.html
+    identifiers = tools.Map(tools.Delegate(ThroughIdentifiers), ctx.embeds.users.data.links.html)
+
+    class Extra:
+        nodes = ctx.embeds.users.data.relationships.nodes.links.related.href
+        locale = ctx.embeds.users.data.attributes.locale
+        date_registered = ctx.embeds.users.data.attributes.date_registered
+        active = ctx.embeds.users.data.attributes.active
+        timezone = ctx.embeds.users.data.attributes.timezone
+        profile_image = ctx.embeds.users.data.links.profile_image
+
+
+class Contributor(Parser):
+    person = tools.Delegate(Person, ctx)
+    order_cited = ctx('index')
+    cited_name = ctx.embeds.users.data.attributes.full_name
+
+
+class Tag(Parser):
+    name = ctx
+
+
+class ThroughTags(Parser):
+    tag = tools.Delegate(Tag, ctx)
+
+
+class Institution(Parser):
+    name = ctx.attributes.name
+    url = ctx.links.self
+
+    class Extra:
+        nodes = ctx.relationships.nodes.links.related.href
+        users = ctx.relationships.users.links.related.href
+        registrations = ctx.relationships.registrations.links.related.href
+        description = ctx.attributes.description
+
+
+class Association(Parser):
+    pass
+
+
+class Link(Parser):
+    url = ctx
+    type = tools.Static('provider')
+
+
+class ThroughLinks(Parser):
+    link = tools.Delegate(Link, ctx)
+
+
+class Subject(Parser):
+    name = ctx.text
+
+
+class ThroughSubjects(Parser):
+    subject = tools.Delegate(Subject, ctx)
+
+
+class Preprint(Parser):
+    title = ctx.attributes.title
+    description = ctx.attributes.description
+    contributors = tools.Map(tools.Delegate(Contributor), ctx['contributors'])
+    institutions = tools.Map(
+        tools.Delegate(Association.using(entity=tools.Delegate(Institution))),
+        ctx.embeds.affiliated_institutions.data
+    )
+    date_updated = tools.ParseDate(ctx.attributes.date_modified)
+    tags = tools.Map(tools.Delegate(ThroughTags), ctx.attributes.category, ctx.attributes.tags)
+    rights = tools.Maybe(ctx, 'attributes.node_license')
+    links = tools.Map(tools.Delegate(ThroughLinks), ctx.links.html)
+    subjects = tools.Map(
+        tools.Delegate(ThroughSubjects),
+        tools.Concat(tools.Static({'text':'Social and behavioral sciences'}))
```

Comment:
Oh weird. I did a similar thing for [bioRxiv](https://github.com/CenterForOpenScience/SHARE/blob/develop/providers/org/biorxiv/normalizer.py#L61) but didn't use a dictionary. Then in the subject class it was just `ctx` instead of `ctx.text`. But as long as it works it doesn't really matter.